### PR TITLE
New data on project page for ISO

### PR DESCRIPTION
**Pull request by ybaras**
#267

### CODE REVIEW

File: productivity/modules/custom/productivity_project/productivity_project.module [hunk 1]
```diff
@@ -802,3 +807,49 @@ function productivity_project_get_developer_chart($node = NULL) {
 
   return $chart_container;
 }
+
+/**
+ * Calculate the total project Budget
```
Comment (ybaras) [inline productivity/modules/custom/productivity_project/productivity_project.module]: ->.
Budget => budget

File: productivity/modules/custom/productivity_project/productivity_project.module [hunk 2]
```diff
@@ -441,9 +441,12 @@ function _productivity_project_build_repo_list() {
...
   $results = array();
 
-  list($public, $secret) = array(variable_get('github_public'), variable_get('github_secret'));
+  list($public, $secret) = array(
+    variable_get('github_public'),
+    variable_get('github_secret')
+  );
   if ($public && $secret) {
-    $query = '?client_id=' . $public . '&client_secret='. $secret . '&page=';
+    $query = '?client_id=' . $public . '&client_secret=' . $secret . '&page=';
```
Comment (amitaibu) [inline productivity/modules/custom/productivity_project/productivity_project.module]: `drupal_http_request()` should get the query string as an array

Comment (amitaibu) [inline productivity/modules/custom/productivity_project/productivity_project.module]: (can be done in a follow up PR).

/cc @nader77

File: productivity/modules/custom/productivity_project/productivity_project.module [hunk 3]
```diff
@@ -379,7 +379,7 @@ function productivity_project_notify($wrapper_project, $arguments) {
  */
 function productivity_project_get_total_scope($project_wrapper, $work_type = FALSE) {
   $total = 0;
-  foreach($project_wrapper->field_table_rate as $rate) {
+  foreach ($project_wrapper->field_table_rate as $rate) {
```
Comment (ybaras) [inline productivity/modules/custom/productivity_project/productivity_project.module]: Fix spacing according to Drupal

File: productivity/modules/custom/productivity_project/productivity_project.module [hunk 4]
```diff
@@ -802,3 +807,56 @@ function productivity_project_get_developer_chart($node = NULL) {
...
+function productivity_project_get_total_budget($wrapper) {
+  $budget = 0;
+  foreach ($wrapper->field_table_rate as $rate) {
+    if (empty($rate->field_rate->value())) {
+      continue;
+    }
+    $current_rate = $rate->field_rate->amount->value();
+    $interval = $rate->field_scope->interval->value();
+    $period = $rate->field_rate_type->value();
+    $budget += ($period == 'hours' || $period == 'days') ? $current_rate * $interval : $current_rate;
```
Comment (ybaras) [inline productivity/modules/custom/productivity_project/productivity_project.module]: add some comment, regarding global.

File: productivity/modules/custom/productivity_project/productivity_project.module [hunk 5]
```diff
@@ -802,3 +807,56 @@ function productivity_project_get_developer_chart($node = NULL) {
...
+    $currency = $rate->field_rate->currency->value();
```
Comment (ybaras) [inline productivity/modules/custom/productivity_project/productivity_project.module]: // Assuming all rates are in the same currency.
 Maybe also create a followup issue to validate all currency are unified.

File: productivity/modules/custom/productivity_project/productivity_project.module [hunk 6]
```diff
@@ -802,3 +807,56 @@ function productivity_project_get_developer_chart($node = NULL) {
...
+  }
+
+  if (!$budget) {
+    return '0';
+  }
+  return number_format($budget) . ' ' . $currency;
+}
+
+/**
+ * Renders the project scope by type
```
Comment (ybaras) [inline productivity/modules/custom/productivity_project/productivity_project.module]: -> dot at the end of sentence.

File: productivity/modules/custom/productivity_project/productivity_project.module [hunk 7]
```diff
@@ -802,3 +807,56 @@ function productivity_project_get_developer_chart($node = NULL) {
...
+ */
+function productivity_project_get_scope($wrapper) {
+  $scopes = [];
```
Comment (ybaras) [inline productivity/modules/custom/productivity_project/productivity_project.module]: $scopes = []; => array();

File: productivity/modules/custom/productivity_project/productivity_project.module [hunk 8]
```diff
@@ -802,3 +807,56 @@ function productivity_project_get_developer_chart($node = NULL) {
...
+  foreach ($wrapper->field_table_rate as $rate) {
+    $current_type = $rate->field_issue_type->value();
+    $current_scope = $rate->field_scope->interval->value();
+    $scopes[$current_type] = (isset($scopes[$current_type])) ? $scopes[$current_type] + $current_scope : $current_scope;
```
Comment (ybaras) [inline productivity/modules/custom/productivity_project/productivity_project.module]: Remove extra code, no need for per Type.

File: productivity/modules/custom/productivity_project/productivity_project.module [hunk 9]
```diff
@@ -802,3 +807,56 @@ function productivity_project_get_developer_chart($node = NULL) {
...
+    $types--;
+    $output_by_type .= $index . ': ' . $item;
+    $output_by_type .= ($types) ? '<br>' : '';
+  }
+
+  return $output_total . ' ' . ucfirst($period) . 's';
+}
+
+/**
+ * Render the start and end dates
```
Comment (ybaras) [inline productivity/modules/custom/productivity_project/productivity_project.module]: -> .

File: productivity/themes/custom/bootstrap_subtheme/templates/node--project.tpl.php [hunk 10]
```diff
@@ -85,6 +85,28 @@
 <div class="row">
   <div class="col-lg-4 col-sm-6 col-xs-12">
     <div class="main-box infographic-box">
+      <i class="fa fa-money red-bg"></i>
+      <span class="headline"><?php print t('Project budget') ?></span>
+      <span class="value"><?php print $total_budget; ?></span>
+
+    </div>
+  </div>
+  <div class="col-lg-4 col-sm-6 col-xs-12">
```
Comment (ybaras) [inline productivity/themes/custom/bootstrap_subtheme/templates/node--project.tpl.php]: Move to left side, bellow actual hours.

File: productivity/themes/custom/bootstrap_subtheme/template.php [hunk 11]
```diff
@@ -25,9 +25,120 @@ function bootstrap_subtheme_preprocess_node__project__full(&$variables) {
...
+  $header = [
+    'Milestone',
+    'Start date',
+    'End date',
+    'Scope'
+  ];
+  $rows = [];
+  $i = 0;
+  foreach ($wrapper->field_timetable_milestones as $milestone) {
+    $rows[$i]['field_milestone'] = $milestone->field_milestone->value();
```
Comment (ybaras) [inline productivity/themes/custom/bootstrap_subtheme/template.php]: $rows[$i] =>  $rows[]

File: productivity/themes/custom/bootstrap_subtheme/template.php [hunk 12]
```diff
@@ -25,9 +25,120 @@ function bootstrap_subtheme_preprocess_node__project__full(&$variables) {
...
+  ];
+  $humanRoles = [
+    'pm' => 'Project manager',
+    'developer' => 'Developer',
+    'qa' => 'Quality assurance (QA)'
+  ];
+  $rows = [];
+  $i = 0;
+  foreach ($wrapper->field_internal_team->value() as $milestone) {
+    $rows[$i]['field_employee'] = $milestone->field_employee['und'][0]['entity']->name;
```
Comment (ybaras) [inline productivity/themes/custom/bootstrap_subtheme/template.php]: same as above.

File: productivity/modules/custom/productivity_project/productivity_project.features.inc [hunk 13]
```diff
@@ -11,6 +11,12 @@ function productivity_project_ctools_plugin_api($module = NULL, $api = NULL) {
   if ($module == "field_group" && $api == "field_group") {
     return array("version" => "1");
   }
+  if ($module == "multifield" && $api == "") {
```
Comment (ybaras) [inline productivity/modules/custom/productivity_project/productivity_project.features.inc]: remove

**Comment by ybaras** [comment]
@bricel Ready for code review. I do the changes from the issue, and fix the tests for the drupal side of things. Because it's already a big PR @nirgn975 told my to open new PR for the front-end tests (It's will be on #280 ).

**Comment by bricel** [comment]
Closed in favor of #281.

**Pull request closed without merging**